quickbudget/views.py

```diff
@@ -0,0 +1,162 @@
+    serializer_class = BudgetSerializer
+
+    def perform_create(self, serializer):
+        # Set the creator field to the current authenticated user
+        serializer.validated_data['creator'] = self.request.user
+
+        if serializer.is_valid():
+            # Create the budget object
+            budget = serializer.save()
+
+            # Add the current user as a member of the budget
+            budget.members.add(self.request.user)
+
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        return Response(serializer.data, status=status.HTTP_400_BAD_REQUEST)
+
+    def get_queryset(self):
+        return Budget.objects.filter(members=self.request.user)
+
+
+class BudgetDetail(generics.RetrieveUpdateDestroyAPIView):
+    queryset = Budget.objects.all()
+    serializer_class = BudgetSerializer
+    lookup_url_kwarg = "budget_id"
+    permission_classes = [BudgetMembersOnly]
+
+
+class AddExpenses(generics.CreateAPIView):
+    queryset = Expense.objects.all()
+    serializer_class = ExpenseSerializer
+    permission_classes = [AllBudgetExpenseMembersOnly]
+
+
+class ListAddExpenses(generics.ListCreateAPIView):
+    serializer_class = ExpenseSerializer
+    permission_classes = [AllBudgetExpenseMembersOnly]
+    filter_backends = (filters.OrderingFilter,)
+    ordering_fields = ["name", "total", "created_timestamp"]
+
+    def perform_create(self, serializer):
+        # Set the creator field to the current authenticated user
+        serializer.validated_data['author'] = self.request.user
+
+        if serializer.is_valid():
+            # Create the budget object
+            budget = serializer.save()
+
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        return Response(serializer.data, status=status.HTTP_400_BAD_REQUEST)
+
+    def get_queryset(self):
+        budget = self.kwargs["budget_id"]
+        queryset = Expense.objects.filter(budget_id=budget).order_by(
+            "created_timestamp")
+        category = self.request.query_params.get("category")
+        if category is not None:
+            queryset = queryset.filter(category=category)
+
+        return queryset
+
+
+class ExpenseDetail(generics.RetrieveUpdateDestroyAPIView):
+    queryset = Expense.objects.all()
+    serializer_class = ExpenseSerializer
+    lookup_url_kwarg = "expense_id"
+
+    permission_classes = [ExpenseBudgetMembersOnly]
+
+
+class ListCategoryList(generics.ListAPIView):
+    queryset = Category.objects.all()
+    serializer_class = CategoriesSerializer
+
+
+class BudgetMembers(APIView):
+    queryset = Budget.objects.all()
+    permission_classes = [BudgetMembersOnly]
+
+
+    def put(self, request, budget_id, format=None):
+        budget = Budget.objects.get(id=budget_id)
+        serializer = MemberAddSerializer(budget, data=request.data)
+        self.check_object_permissions(request, budget)
+
+        if self.request.user.id == budget.creator_id:
```

Comment:
better to update to the following:
```
if self.request.user.id != budget.creator_id:
    return Response(serializer.errors, status=status.HTTP_403_FORBIDDEN)
```

in this case you don't need to put the code inside the `if` block.